### PR TITLE
Route and IP Address assignment propagation to ztcentral from terraform / Pinging between two docker containers

### PR DESCRIPTION
**Pull request by erikh**
This allows terraform to propagate routes and assigned ip blocks with the following example:

More is coming here and I expect to be able to ping things when I am done.

Also included is the previous unmerged as-of-yet PR, which can be merged
without impacting this one.

```hcl
terraform {
  required_providers {
    docker = {
      source = "kreuzwerker/docker"
    }
    zerotier = {
      source  = "zerotier.com/dev/zerotier"
      version = "0.2"
    }
  }
}

provider "docker" {}

resource "docker_network" "private_network" {
  name = "my_network"
}

resource "docker_image" "zerotier" {
  name         = "bltavares/zerotier"
  keep_locally = true
}

resource "docker_container" "uno" {
  name  = "zerotier_uno"
  image = docker_image.zerotier.latest
}

resource "docker_container" "dos" {
  name  = "zerotier_dos"
  image = docker_image.zerotier.latest
}


#
# Provider
#

provider "zerotier" {}

#
# Alice
#

variable "ipv4_cidr" {
  default = "10.0.1.0/24"
}

resource "zerotier_network" "docker_network" {
  name = "docker"
  assignment_pool = [{
    start = "10.0.1.2"
    end   = "10.0.1.253"
  }]

  routes = [{
    target = var.ipv4_cidr
    via    = "10.0.0.1"
  }]
}

resource "zerotier_identity" "alice" {}

resource "zerotier_member" "alice" {
  name       = "docker-alice"
  node_id    = zerotier_identity.alice.id
  network_id = zerotier_network.docker_network.id
}

resource "zerotier_identity" "bob" {}

resource "zerotier_member" "bob" {
  name       = "bob"
  node_id    = zerotier_identity.bob.id
  network_id = zerotier_network.docker_network.id
}

```

Signed-off-by: Erik Hollensbe <github@hollensbe.org>

### CODE REVIEW

File: pkg/zerotier/resource_network.go [hunk 1]
```diff
@@ -32,15 +35,126 @@ func resourceNetwork() *schema.Resource {
...
+	}
+
+	var last net.IP
+
+	prefixLen, bits := nw.Mask.Size()
+
+	if prefixLen == bits {
+		last = first
+	} else {
+		val := big.NewInt(0)
```
Comment (erikh) [inline pkg/zerotier/resource_network.go:71]: one thing I wanted to ask you @someara in particular was about this code... it ends up calculating offsets by 1 intentionally, e.g.: 10.0.1.0/24 will be first ip as 10.0.1.2 and last as 10.0.1.254

is this desired behavior from this library or should I just make it return .0 and .255?

**Comment by erikh** [comment]
Don't sweat the document structure too hard; I expect it to change as we move forward.

File: pkg/zerotier/resource_network.go (hunk 1 above)
Comment (someara) [reply pkg/zerotier/resource_network.go]: Let's just do 0 and 255

Comment (erikh) [reply pkg/zerotier/resource_network.go]: I ended up with just specifying the start/end over CIDR. I can add CIDRs back optionally if you want.

**Comment by erikh** [comment]
This finishes the job and adds a docker ping test. I think the WIP state can be removed, but I don't have a good way to test routes yet.

**Comment by erikh** [comment]
Note that a lot has changed here. We moved to a different image which has a different entrypoint to better accommodate terraform as mentioned elsewhere.

The ping test includes a lot of functions I think we can re-use later for more integration/acceptance testing.

**Comment by erikh** [comment]
Sorry this PR might need to be broken up, let's discuss it when you are around. That said, `make test` should work if you pull it and setup an API key.

**Pull request merged**